mesonbuild/minstall.py

```diff
@@ -75,20 +77,27 @@ def add_arguments(parser: argparse.Namespace) -> None:
                         help='Do not print every file that was installed.')
     parser.add_argument('--destdir', default=None,
                         help='Sets or overrides DESTDIR environment. (Since 0.57.0)')
+    parser.add_argument('--dry-run', '-n', action='store_true',
+                        help='Doesn\'t actually install, but print logs.')
 
 class DirMaker:
-    def __init__(self, lf: T.TextIO):
+    def __init__(self, lf: T.TextIO, makedirs):
```

Comment:
This is a nasty corner case of typing, passing callables with keyword arguments. the best you can do here (right now) is:
```python
T.Callable[..., None]
```